example/example/spiders/books_06.py

```diff
@@ -12,34 +12,34 @@
 import scrapy
 import attr
 
-from scrapy_po import WebPage, ItemPage
+from scrapy_po import WebPage, ItemWebPage, Injectable
 
 
-class ListingsPiece(WebPage):
+class ListingsExtractor(WebPage):
     def urls(self):
         return self.css('.image_container a::attr(href)').getall()
 
 
-class PaginationPiece(WebPage):
+class PaginationExtractor(WebPage):
     def urls(self):
         return self.css('.pager a::attr(href)').getall()
 
 
-class BreadcrumbsPiece(WebPage):
+class BreadcrumbsExtractor(WebPage):
     def urls(self):
         return self.css('.breadcrumb a::attr(href)').getall()
 
 
 @attr.s(auto_attribs=True)
-class ListingsPage(WebPage):
-    book_list: ListingsPiece
-    pagination: PaginationPiece
+class ListingsPage(Injectable):
```

Comment:
Note this class is a "Page Object" (i.e. the suffix in the name is "Page" because it extracts all that can be extracted from a listings page), but that it is not implementing the `to_item` (it was not required in the code), so it is extending `Injectable`. Probably not very easy to understand. 

By the other side it looks nice that it can be just the composition of two other components. This is an alternative to the monolithc version proposed in https://gist.github.com/kmike/2fa2b1b8c77bfb98f72b2636e66c5d36#id2 with the name of `BookListPage`.